main_test.go

```diff
@@ -535,7 +535,10 @@ func TestEncodeNil(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if bytes.Equal(buf, []byte("$-1\r\n")) == false {
+	// Redis has two kinds of nil object: "Null Bulk String" representing as
+	// "$-1\r\n" and "Null Array" representing as "*-1\r\n".
+	// The encode result of nil returns "-1\r\n" only.
+	if bytes.Equal(buf, []byte("-1\r\n")) == false {
```

Comment:
How is this going to be interpreted by a client? `-1\r\n` is not defined by the redis protocol and could produce undefined behaviour.